### ISSUE TITLE
Fire TV/Fire Stick support

### ISSUE THREAD

**Issue by nascentt** [mvysny/photocloud-frame-slideshow#49]
Please please please add Amazon Fire TV/TV Stick support for this app, a photo screensaver from our own photos would be so good on the large screen.  I'd be more than willing to help with any testing

**Comment by mvysny** [comment]
Thanks ;) Probably not happening in the near future since I don't have the TV stick. Now if I understand correctly, Fire TV is a TV running some Amazon Android, so you can install PhotoCloud from the Amazon Underground App Store and run it there. Is this correct? I'm just guessing since I don't have the Fire TV.

The TV stick is something entirely different - it's a HDMI stick pluggable into any TV (like Google Chromecast) which can be controlled from a phone. While looking for info whether it supports remote display, I found this: https://www.amazon.com/gp/help/customer/display.html?nodeId=201453020
Basically, you let your phone run the Photocloud, and just mirror the phone screen to the stick and thus to your TV. Is this possible for you?

**Comment by mvysny** [comment]
It seems that I would have to develop the app specifically for Fire TV and since I can't purchase the TV in Finland this is impossible: https://developer.amazon.com/docs/fire-tv/faq-general.html

**Comment by mvysny** [comment]
I am unable to find any information on whether Fire Stick supports remote display which is basically a prerequisite for PhotoCloud to work. All I can find is a `RemoteMediaPlayer` class which is not enough since it only can show a photo downloaded off an URL. That may be possible for some cloud storages, but definitely not possible to show images from cache, or from your local samba share.

It looks like supporting Fire is just not possible, I'm sorry.

**Comment by nascentt** [comment]
So I wasn't even aware of a Fire TV all in one device (amazon built into a tv) I was only aware of the Fire TV Stick (which I have) and the Fire TV set top box (which I don't currently have but plan to buy when it reduces in price).

Personally the mirroring of a phone doesnt suit me at all as the pictures are on a computer (file server) and would need to be played without the phone in the house (multiple people live here) but that being said, if that's lal that can happen then I will have to buy a tablet to pair with it to mirror.

Is there anything I can check to see if the fire stick supports remote play? The need to stream from a samba server was 100% my desire for it so if this is true then I'll be very disappointed.

If needs be i'll try and buy a fire stick to send you (wish I thought of it prior to black friday as it was cheap).

**Comment by mvysny** [comment]
> Personally the mirroring of a phone doesnt suit me at all as the pictures are on a computer (file server) and would need to be played without the phone in the house (multiple people live here) but that being said, if that's lal that can happen then I will have to buy a tablet to pair with it to mirror.

True - if you want to stream those photos even when you and your phone are outside, you need an additional device for mirroring. But I am under an impression that you can't actually install custom apps on the Fire Stick itself (similarly to Google Chromecast - you can't install any custom apps there), so there is always a need for additional device, streaming photos to the Stick.

> Is there anything I can check to see if the fire stick supports remote play? The need to stream from a samba server was 100% my desire for it so if this is true then I'll be very disappointed.

Thanks for the help but I don't think there is anything you can do. I need to find a programmatic way of controlling that stick (the API), so I'll try to google more ;) Maybe you could try to find an app that actually streams photos to the Stick - that way we would have a proof that such way of streaming is possible.

> If needs be i'll try and buy a fire stick to send you (wish I thought of it prior to black friday as it was cheap).

Thanks, that's an awesome offer ;) Yet no worries - I can buy the Stick from amazon.com if need be. Also, first we need to discover that remote control API.

**Comment by nascentt** [comment]
The only thing I've really been able to find that doesn't do it based on 'the cloud' is this https://www.amazon.co.uk/DS-photo-for-Fire-TV/dp/B006ZCV46G but I don't have a synology NAs to test it with.

Anything else seems to just be a streamer https://www.amazon.co.uk/Dipendu-Saha-AirBuddy/dp/B00JHAXIIA  + https://www.amazon.co.uk/Verbatim-Americas-LLC-MediaShare/dp/B00EYC8P3E unless I'm misreading.

**Comment by mvysny** [comment]
Yeah, it looks like a streamer app. And even that Synology nas app can actually stream the photos as well, from the nas via http via your local lan.

The Google Chromecast allows photocloud to configure mirroring and thus air the photos from the phone as they are shown on your display. I guess fire stick does not allow that. 

I could maybe turn photocloud into some kind of a http server, offering photos... Maybe... Hard to say.

**Comment by nascentt** [comment]
I appreciate the effort you've put into considering it.  I'm naively
thought if amazon fire tablets could do it, so could a fire stick.

**Comment by mvysny** [comment]
Yeah. Sorry to let you down. I guess there's nothing more we can do.

**Comment by mvysny** [comment]
Closing as won't fix.
